Sources/AWSLambdaRuntimeCore/DetachedTasks.swift

```diff
@@ -0,0 +1,169 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import Foundation
+import NIOConcurrencyHelpers
+import NIOCore
+import Logging
+
+/// A container that allows tasks to finish after a synchronous invocation
+/// has produced its response.
+public final class DetachedTasksContainer {
+
+    struct Context {
+        let eventLoop: EventLoop
+        let logger: Logger
+    }
+    
+    private var context: Context
+    private var storage: Storage
+
+    init(context: Context) {
+        self.storage = Storage()
+        self.context = context
+    }
+
+    /// Adds a detached task that runs on the given event loop.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the task.
+    ///   - task: The task to execute. It receives an `EventLoop` and returns an `EventLoopFuture<Void>`.
+    /// - Returns: A `RegistrationKey` for the registered task.
+    @discardableResult
+    public func detached(name: String, task: @escaping (EventLoop) -> EventLoopFuture<Void>) -> RegistrationKey {
+        let key = RegistrationKey()
+        let task = task(context.eventLoop).always { _ in
+            self.storage.remove(key)
+        }
+        self.storage.add(key: key, name: name, task: task)
+        return key
+    }
+    
+    /// Adds a detached async task.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the task.
+    ///   - task: The async task to execute.
+    /// - Returns: A `RegistrationKey` for the registered task.
+    @discardableResult
+    public func detached(name: String, task: @Sendable @escaping () async throws -> Void) -> RegistrationKey {
```

Comment:
We should enforce that users handle errors! For this reason the user's closure should just be `async` but not `throws`

Reply:
Definitely the best approach 👍🏻